source/extensions/quic_listeners/quiche/active_quic_listener_config.cc

```diff
@@ -0,0 +1,27 @@
+#include "extensions/quic_listeners/quiche/active_quic_listener_config.h"
+
+#include "envoy/api/v2/listener/quic_config.pb.h"
+
+#include "server/well_known_names.h"
+
+#include "extensions/quic_listeners/quiche/active_quic_listener.h"
+
+namespace Envoy {
+namespace Quic {
+
+ProtobufTypes::MessagePtr ActiveQuicListenerConfigFactory::createEmptyConfigProto() {
+  return std::make_unique<envoy::api::v2::listener::QuicProtocolOptions>();
+}
+
+Network::ActiveUdpListenerFactoryPtr
+ActiveQuicListenerConfigFactory::createActiveUdpListenerFactory(const Protobuf::Message& message) {
+  auto& config = dynamic_cast<const envoy::api::v2::listener::QuicProtocolOptions&>(message);
+  return std::make_unique<ActiveQuicListenerFactory>(config);
+}
+
+std::string ActiveQuicListenerConfigFactory::name() { return Server::UdpListenerNames::get().Quic; }
```

Comment:
The name can now be moved out of `Server::UdpListenerNames` and just inlined here? I don't think it needs to live outside the extension? (If it does please move it into an extension well known names file)

Reply:
I'm not sure what inline means here. The string is also used in tests. So I added a statistic const in the .h file.